exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -143,6 +147,25 @@ func newSaramaProducer(config Config) (sarama.SyncProducer, error) {
 	c.Producer.MaxMessageBytes = config.Producer.MaxMessageBytes
 	c.Producer.Flush.MaxMessages = config.Producer.FlushMaxMessages
 
+	for _, u := range []string{os.Getenv("HTTP_PROXY"), os.Getenv("HTTPS_PROXY")} {
+		if u != "" {
+			httpProxyURI, err := url.Parse(u)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			httpDialer, err := proxy.FromURL(httpProxyURI, proxy.Direct)
+			c.Net.Proxy = struct {
+				Enable bool
+				Dialer proxy.Dialer
+			}{
+				Enable: true,
+				Dialer: httpDialer,
+			}
+			break
```

Comment:
What happens if both env vars are set?

Can you include a line in README about this?

Is there a way to have a unit test to check this behavior?

Reply:
I saw somewhere in the docs that any `http(s)_proxy` envs set would be picked up but not in the case of sarama it seem, that is why I went with this route.  
It might be a better approach to have an env specific to sarama like `SARAMA_PROXY` as you are speaking Kafka, not http, so not a typical scenario. That or some optional arg you can set in the yaml. The latter is how I use it in a consumer I created, I have a `WithProxy` option which sets a proxy field on a struct. If set I do the above.

Reply:
I am using something along these lines:

Reply:
Please don't reference variables fetched from the operating operating environment, it complicates how to configure the exporter further and can't easily be caught in within the config validation to save time providing feedback to the user (if it is set incorrectly).